source/common/stats/thread_local_store.cc

```diff
@@ -75,6 +91,29 @@ void ThreadLocalStoreImpl::shutdownThreading() {
   shutting_down_ = true;
 }
 
+void ThreadLocalStoreImpl::mergeHistograms(PostMergeCb mergeCb) {
+  if (!shutting_down_) {
+    tls_->runOnAllThreadsWithBarrier(
+        [this]() -> void {
+          for (ScopeImpl* scope : scopes_) {
+            for (auto histogram : tls_->getTyped<TlsCache>().scope_cache_[scope].histograms_) {
+              histogram.second->beginMerge();
+            }
+          }
+        },
+        [this, mergeCb]() -> void { mergeInternal(mergeCb); });
+  }
+}
+
+void ThreadLocalStoreImpl::mergeInternal(PostMergeCb mergeCb) {
+  if (!shutting_down_) {
+    for (auto histogram : histograms()) {
```

Comment:
explicit type here & other range loops where the type is not locally obvious.

Reply:
for range based loops on maps, looks like auto style c++ 11 is being used across the code. Do you want to change this here using iterator?

Reply:
Range-based for-loops are great.  Just asking you to be specific with the type of the loop var when not obvious, which IMO is the case here.  See the Google C++ style guide (applicable in Envoy with exceptions): https://google.github.io/styleguide/cppguide.html#auto)

There may be other cases in Envoy where 'auto' is used in lieu of a non-obvious type as well, but I'm only commenting on the code in this PR and the style guide.

One practical value of being explicit is that then I can see more clearly whether you are copying the object or referencing it, which is non-obvious with auto.

I am not asking you to use an explicit iterator object, though in another comment I noted that iterator variables usually have obvious type, and are thus appropriate for 'auto'.

Reply:
ok. Thanks. Got it. I changed this specific case.